lib/Dialect/TTMetal/Transforms/Passes.cpp

```diff
@@ -655,111 +656,238 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
-          iterators[blockArgIteratorMapping[1]], dstIndex);
-    } else if (mlir::isa<arith::MulFOp>(op)) {
-      builder.create<ttkernel::MulTilesOp>(
-          op.getLoc(), cbOperands[operandIndices[0]],
-          cbOperands[operandIndices[1]], iterators[blockArgIteratorMapping[0]],
-          iterators[blockArgIteratorMapping[1]], dstIndex);
+                        Value dstTileIndex, OpBuilder &builder,
+                        std::int64_t numDpsInputs) const {
+
+    if (numDpsInputs == 1) {
+      convertComputeUnaryOp(arithOrMathOp, cbOperands, iterators,
+                            blockArgIteratorMapping, dstTileIndex, builder);
+    } else if (numDpsInputs == 2) {
+      convertComputeBinaryOp(arithOrMathOp, cbOperands, iterators,
+                             blockArgIteratorMapping, dstTileIndex, builder);
     } else {
-      llvm_unreachable("Unhandled conversion");
+      llvm_unreachable("Unhandled conversion for operation which is neither "
+                       "unary nor binary.");
     }
   }
 
+  // Builds instructions to execute before looping over tiles has started.
+  void buildInitSection(Operation &arithOrMathOp,
+                        OpBuilder &dispatchBlockBuilder,
+                        ArrayRef<BlockArgument> cbOperands,
+                        std::int64_t numDPSInputs) const {
+    convertComputeInitOp(arithOrMathOp, cbOperands, numDPSInputs,
+                         dispatchBlockBuilder);
+  }
+
+  // Builds nested loops which loop over tensor tiles after initalization is
+  // done and computation to perform on each tile over which loops iterate.
+  void buildLoopsAndComputation(Operation &arithOrMathOp,
+                                OpBuilder &dispatchBlockBuilder,
+                                ArrayRef<BlockArgument> &cbOperands,
+                                std::int64_t numDPSInputs) const {
+    // Create loops which iterate over tiles in tensor.
+    LoopNest loopNest =
+        createLoopNest(cbOperands, numDPSInputs, dispatchBlockBuilder);
+    assert(loopNest.loops.size() == 2 && "Expected only two loops!");
+
+    // The loop nest is created from outermost to innermost. Get the inner loop
+    // and place computation calls inside it.
+    Region *innerLoopRegion = loopNest.loopRegions.back();
+    const Location &location = innerLoopRegion->getLoc();
+    ArrayRef<BlockArgument> iterators =
+        loopNest.loops.back().getRegionIterArgs();
+    SmallVector<unsigned> blockArgIteratorMapping =
+        loopNest.blockArgIteratorMapping;
+
+    OpBuilder innerLoopBuilder(&innerLoopRegion->front(),
+                               innerLoopRegion->front().begin());
+
+    // We always operate on the first and only tile in DST register.
+    Value dstTileIndex = i32(0, innerLoopBuilder);
+    auto outCBTileIndex = iterators[blockArgIteratorMapping.back()];
+    auto outCB = cbOperands.back();
+
+    // MATH acquires lock on DST register.
```

Comment:
Do we need these comments here? I feel it is not telling anything new that description of the `ttkernel::TileRegsAcquireOp` is not saying already. I think it would be more valuable to add a comment why are we doing acquire op at this place, or simply delete the comment. Not to leave bunch of review comments, same applies for some other kernel ops in this function